crates/matrix-sdk-crypto/src/types/one_time_keys.rs

```diff
@@ -110,24 +110,28 @@ pub enum OneTimeKey {
 }
 
 impl OneTimeKey {
-    /// Deserialize the [`OneTimeKey`] from a [`DeviceKeyAlgorithm`] and a Raw
+    /// Deserialize the [`OneTimeKey`] from a [`OneTimeKeyAlgorithm`] and a Raw
     /// JSON value.
     pub fn deserialize(
-        algorithm: DeviceKeyAlgorithm,
+        algorithm: OneTimeKeyAlgorithm,
         key: &Raw<ruma::encryption::OneTimeKey>,
     ) -> Result<Self, serde_json::Error> {
         match algorithm {
-            DeviceKeyAlgorithm::Curve25519 => {
-                let key: String = key.deserialize_as()?;
-                Ok(OneTimeKey::Key(
-                    Curve25519PublicKey::from_base64(&key).map_err(serde::de::Error::custom)?,
-                ))
-            }
-            DeviceKeyAlgorithm::SignedCurve25519 => {
+            OneTimeKeyAlgorithm::SignedCurve25519 => {
                 let key: SignedKey = key.deserialize_as()?;
                 Ok(OneTimeKey::SignedKey(key))
             }
-            _ => Err(serde::de::Error::custom(format!("Unsupported key algorithm {algorithm}"))),
+            _ => match algorithm.as_str() {
+                "curve25519" => {
```

Comment:
The Matrix spec only defines `signed_curve25519` for one-time keys so we don't have other variants currently in `OneTimeKeyAlgorithm` and we need to fallback to string matching if we want to keep this branch.

Reply:
Do we actually need to keep it for backwards compatibility? Does someone from the crypto team know?

Reply:
The `Key` variant is actually unused, so let's just remove it.

Reply:
Maybe this can be done in a separate PR? Because this is a breaking change for the crypto crate since the enum is public.

Reply:
I guess we can do that as well.